src/ansi-c/library/stdlib.c

```diff
@@ -115,6 +115,9 @@ inline void *malloc(__CPROVER_size_t malloc_size)
   // realistically, malloc may return NULL,
   // and __CPROVER_allocate doesn't, but no one cares
   __CPROVER_HIDE:;
+  if(__CPROVER_malloc_may_fail && __VERIFIER_nondet___CPROVER_bool())
+    return (void *)0;
```

Comment:
1) It may be more consistent to just declare a local, uninitialised variable of type `__CPROVER_bool` to avoid the side-effect of a function call.
2) The non-determinism also needs to be introduced in `calloc`.
3) `realloc` now needs to handle the case of a failing `malloc`.

Reply:
Add 1. No can do: using uninitialised variables doesn't even compile with the current flag set. I can initialise it to `nondet_bool`.

Reply:
2,3 are done.